File: app/src/main/kotlin/com/saatxi/eatapp/ui/common/ExportOptionsDialog.kt
package com.saatxi.eatapp.ui.common

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.saatxi.eatapp.R

/**
 * The "what goes in the file?" step shown before any share or export — one
 * restaurant from the detail screen, the whole list from "share all", or the
 * settings "export my data" row. All three want the same decision (whether to
 * carry visits along), so they share one dialog rather than three hand-rolled
 * variants whose wording could drift.
 *
 * "Include visits" defaults to on: visits are the user's own history and the
 * whole point of a backup, so the safe default is to keep them — turning the
 * switch off is the deliberate opt-out. [onConfirm] receives the chosen value.
 */
@Composable
fun ExportOptionsDialog(
    onConfirm: (includeVisits: Boolean) -> Unit,
    onDismiss: () -> Unit
) {
    var includeVisits by remember { mutableStateOf(true) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text(stringResource(R.string.export_dialog_title)) },
        text = {
            Column {
                Text(stringResource(R.string.export_dialog_body))
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.padding(top = 12.dp)
                ) {
                    Switch(checked = includeVisits, onCheckedChange = { includeVisits = it })
                    Text(
                        text = stringResource(R.string.export_dialog_include_visits),
                        modifier = Modifier.padding(start = 12.dp)
                    )
                }
            }
        },
        confirmButton = {
            TextButton(onClick = { onConfirm(includeVisits) }) {
                Text(stringResource(R.string.action_export))
            }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) {
                Text(stringResource(R.string.action_cancel))
            }
        }
    )
}

File: lib/features/import_export/export_options_dialog.dart
import 'package:flutter/material.dart';

import '../../core/l10n/generated/app_localizations.dart';

/// The "what goes in the file?" step shown before any share or export — one
/// restaurant from the detail screen, the whole list from "share all", or the
/// settings "export my data" row. All three want the same decision (whether to
/// carry visits along), so they share one dialog rather than three hand-rolled
/// variants whose wording could drift.
///
/// Ported from `ui/common/ExportOptionsDialog.kt`. Resolves to the chosen
/// "include visits" value, or null when the user backs out — dismissing with the
/// Cancel button or the system back gesture. Callers read that null as "don't
/// export anything", so a stray tap never writes a file.
Future<bool?> showExportOptionsDialog(BuildContext context) {
  return showDialog<bool>(
    context: context,
    builder: (BuildContext context) => const _ExportOptionsDialog(),
  );
}

class _ExportOptionsDialog extends StatefulWidget {
  const _ExportOptionsDialog();

  @override
  State<_ExportOptionsDialog> createState() => _ExportOptionsDialogState();
}

class _ExportOptionsDialogState extends State<_ExportOptionsDialog> {
  /// On by default: visits are the user's own history and the whole point of a
  /// backup, so turning the switch off is the deliberate opt-out.
  bool _includeVisits = true;

  @override
  Widget build(BuildContext context) {
    final AppLocalizations l10n = AppLocalizations.of(context);
    return AlertDialog(
      title: Text(l10n.exportDialogTitle),
      content: Column(
        mainAxisSize: MainAxisSize.min,
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          Text(l10n.exportDialogBody),
          const SizedBox(height: 12),
          Row(
            children: <Widget>[
              Switch(
                value: _includeVisits,
                onChanged: (bool value) =>
                    setState(() => _includeVisits = value),
              ),
              const SizedBox(width: 12),
              Expanded(child: Text(l10n.exportDialogIncludeVisits)),
            ],
          ),
        ],
      ),
      actions: <Widget>[
        TextButton(
          onPressed: () => Navigator.of(context).pop(),
          child: Text(l10n.actionCancel),
        ),
        TextButton(
          onPressed: () => Navigator.of(context).pop(_includeVisits),
          child: Text(l10n.actionExport),
        ),
      ],
    );
  }
}
